src/components/ERC20Transfers/ERC20Transfers.tsx

```diff
@@ -0,0 +1,318 @@
+// import React from "react";
+// import { useMoralis } from "react-moralis";
+// import { getEllipsisTxt } from "../../helpers/formatters";
+// import { getExplorer } from "../../helpers/networks";
+import "antd/dist/antd.css";
+import { useERC20Transfers } from "../../hooks/useERC20Transfers";
+import Table from "../reusable/Table";
+// import { limitDigits } from "../../helpers/formatters";
+// import { getChainDetails } from "../../helpers/getChainDetails";
+// import { CustomImg } from "../reusable/CustomImg";
+import styled from "styled-components";
+import Blockie from "../Blockie";
+
+import React, { useState, useEffect } from "react";
+import { utils } from "ethers";
+import { fetchNftData } from "./utils";
+
+interface IStyled {
+  top?: string;
+  bottom?: string;
+  left?: string;
+  right?: string;
+  width?: string;
+}
+
+const AbsoluteImgContainer = styled("div")`
+  position: absolute;
+  top: -0.3125rem;
+  top: ${(props: IStyled) => props.top};
+  bottom: ${(props) => props.bottom};
+  left: ${(props) => props.left};
+  right: ${(props) => props.right};
+  width: ${(props) => props.width};
+`;
+
+function ERC20Transfers() {
+  const { ERC20Transfers, chainId } = useERC20Transfers();
+
+  // new implement
+  const [fetchData, setFetchData] = useState({});
+
+  const handleData = async () => {
+    const data = await fetchNftData();
+    setFetchData(data);
+  };
+
+  useEffect(() => {
+    handleData();
+  }, [fetchData]);
```

Comment:
@karangorania, As ``fetchData`` is being updated in the ``handleData`` function, parsing ``fetchData`` in the useEffect's array creates an infinite loop.

Removing ``fetchData`` from the array stops the inifinite loop from happening, but it also means the data won't be updated if any new transactions are made (unless the user refreshes, of course).

To fix this bug and avoid the above scenario, I'd do the following:
- [ ] Store data in a seperate state variable
- [ ] Only update ``fetchData`` once data has been successfuly fetched